packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+        choices: ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN'],
+      };
+      var exp1 = new CronBuilder();
+      const weekAnswer = await inquirer.prompt([WeekQuestion]);
+      const timeQuestion = {
+        type: 'datetime',
+        name: 'dt',
+        message: 'Select the start time (use arrow keys)?',
+        format: ['HH', ':', 'mm', ' ', 'A'],
+      };
+      const timeAnswer = await inquirer.prompt([timeQuestion]);
+      exp1.set(
+        'minute',
+        (timeAnswer.dt as any)
+          .getMinutes()
+          .toString()
+          .split(),
+      );
+      exp1.set(
+        'hour',
+        (timeAnswer.dt as any)
+          .getHours()
+          .toString()
+          .split(),
+      );
+      exp1.set('dayOfTheWeek', Days[(weekAnswer as any).week].split());
+      cloudwatchRule = exp1.build();
+      cloudwatchRule = 'cron(' + cloudwatchRule + ' ' + '*' + ')';
+      break;
+    }
+    case 'monthly': {
+      const dateQuestion = {
+        type: 'datetime',
+        name: 'dt',
+        message: 'Select on which day of the month to invoke the function (dd)(use arrow keys):',
+        format: ['DD'],
+      };
+      const timeQuestion = {
+        type: 'datetime',
+        name: 'dt',
+        message: 'Select the start time (use arrow keys)?',
+        format: ['HH', ':', 'mm', ' ', 'A'],
+      };
+      const timeAnswer = await inquirer.prompt([timeQuestion]);
+      const dateAnswer = await inquirer.prompt([dateQuestion]);
+      cloudwatchRule = makeCron(intervalAnswer.interval, dateAnswer, timeAnswer);
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case 'yearly': {
+      const dateQuestion = {
+        type: 'datetime',
+        name: 'dt',
+        message: 'Select the month and date to invoke the function (mm / dd) (use arrow keys):',
+        format: ['MM', '/', 'DD'],
+      };
+      const timeQuestion = {
+        type: 'datetime',
+        name: 'dt',
+        message: 'Select the start time (use arrow keys)?',
+        format: ['HH', ':', 'mm', ' ', 'A'],
+      };
+      const timeAnswer = await inquirer.prompt([timeQuestion]);
+      const dateAnswer = await inquirer.prompt([dateQuestion]);
+      cloudwatchRule = makeCron(intervalAnswer.interval, dateAnswer, timeAnswer);
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+
+      break;
+    }
+    case 'customRule': {
+      const customRuleQuestion = {
+        type: 'input',
+        name: 'customRule',
+        message: 'Custom Schedule expression(Learn more : https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html)',
+      };
+      const customRuleAnswer = await inquirer.prompt([customRuleQuestion]);
+      cloudwatchRule = 'cron(' + customRuleAnswer.customRule + ')';
+
+      break;
+    }
+  }
+  // check if the given cron is valid
```

Comment:
should there be some validation here?

Reply:
will update in next commit, working on it